#52021 - added deletion of uploaded file. Redesigned file upload button.

ManuBoca92 opened the pull request.

Commit:
#52021 - added deletion of uploaded file. Redesigned file upload button.

bartdominiak commented on examples/now/pages/index.vue:
```diff
@@ -66,9 +66,14 @@
               </b-upload>
               <span
                 v-if="file"
```
I think we should hide "Click to upload" when user provide file (if it support only one file attachment)

```vue
...
<b-upload
  v-if="!file"
  :name="field.name"
  v-model="file"
  :required="field.required"
>
  <a class="tag is-secondary">
    <b-icon icon="upload" />
    <span>
      Click to upload
    </span>
  </a>
</b-upload>
<span
  v-else
  class="tag is-primary"
>
  {{ file.name }}
  <button
    @click="deleteFile()"
    class="delete is-small"
    type="button"
  />
</span>
...
```

ManuBoca92 replied:
I used `v-show="!file"` instead since `v-if="!file"` prevents uploaded file from being sent.

bartdominiak replied:
After that you can add `v-else` instead of `v-if="file"`

bartdominiak replied:
Yeah, You're right. You need to use v-show to keep this `input` value in form.

Keep this in mind. In Javascript we have:
```javascript
const test = null
test === true // return false
test === false // return false
!test === true // return true
!test === false // return false
```
So, `v-if="file"` will return true only if file is added. If it's `null` it wont render anything

bartdominiak commented on examples/now/pages/index.vue:
```diff
@@ -135,6 +140,9 @@ export default {
     },
+    deleteFile() {
+      this.file = null
+    },
```
IMO it's unecessary to add another method, you can just add
```vue
<button @click="file = null" />

ManuBoca92 replied:
Done.

ManuBoca92 committed:
52021 - Fix CR.

Removed redundant method which set file object to null and added a conditional statement  for rendering file object.
